GANDLF/utils/generic.py

```diff
@@ -86,23 +88,6 @@ def get_filename_extension_sanitized(filename: str) -> str:
     return ext
 
 
-def parse_version(version_string: str) -> int:
-    """
-    Parses version string, discards last identifier (NR/alpha/beta) and returns an integer for comparison.
-
-    Args:
-        version_string (str): The string to be parsed.
-
-    Returns:
-        int: The version number.
-    """
-    version_string_split = version_string.replace("-dev", "")
-    version_string_split = version_string_split.split(".")
-    if len(version_string_split) > 3:
-        del version_string_split[-1]
-    return int("".join(version_string_split))
-
-
 def version_check(version_from_config: str, version_to_check: str) -> bool:
```

Comment:
Can you plz fix typing here also (is it dict, right?)
```suggestion
def version_check(version_from_config: dict[str, str], version_to_check: str) -> bool:
```

Reply:
Hm
I see you imply `version_from_config["minimum"]` can be int. Is it really so? Looks like neither old code (`parse_version` function) nor new (`Version`) cannot parse ints:
```
In [4]: Version(1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 Version(1)
....
    197 if not match:
    198     raise InvalidVersion(f"Invalid version: '{version}'")

TypeError: expected string or bytes-like object
```